File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxsub = nums[0]
        cur = 0
        for i in nums:
            if cur < 0:
                cur = 0
            cur += i
            maxsub = max(cur, maxsub)
        return maxsub

File: 53-maximum-subarray/maximum-subarray.ts
function maxSubArray(nums: number[]) {
    let maxarray:number=-76996985346
    let minsum:number=0
    for (let i = 0; i < nums.length; i++) {
        minsum=minsum+nums[i]
if (minsum>maxarray) {
    maxarray=minsum
}
if (minsum<0) {
    minsum=0
}        
    }
    return maxarray
};
